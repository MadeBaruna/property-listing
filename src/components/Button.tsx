import React from 'react';
import styled from 'styled-components/macro';

const ButtonStyled = styled.button`
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  transition: 200ms ease-out;

  &.outlined {
    background: white;
    border: 1px solid #04ACF4;
    color: #04ACF4;

    &:hover {
      background: #04ACF4;
      color: white;
    }
  }

  &.disabled {
    background: grey;
  }

  &:hover {
    background: #0484f4;
    transition: 200ms ease-out;

    &.disabled {
      background: grey;
      cursor: default;
    }
  }
`;

interface IProps extends React.HTMLProps<HTMLButtonElement> {
  outlined?: boolean;
  disabled?: boolean;
}

const Button: React.FC<IProps> = ({ children, outlined, disabled, style, onClick }) => (
  <ButtonStyled
    style={style}
    onClick={disabled ? () => { } : onClick}
    className={`${outlined ? 'outlined' : ''} ${disabled ? 'disabled' : ''}`}>
    {children}
  </ButtonStyled>
);

export default Button;
