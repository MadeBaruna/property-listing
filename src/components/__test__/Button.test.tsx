import React from 'react';
import 'jest-styled-components'
import { render } from 'enzyme';

import Button from '../Button';

it('renders <Button />', () => {
  const wrapper = render(
    <Button />
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders <Button /> with style', () => {
  const wrapper = render(
    <Button style={{ float: "right" }} />
  );

  expect(wrapper).toMatchSnapshot();
});
