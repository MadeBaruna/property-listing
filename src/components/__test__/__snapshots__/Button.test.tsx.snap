// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Button /> 1`] = `
.c0 {
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.outlined {
  background: white;
  border: 1px solid #04ACF4;
  color: #04ACF4;
}

.c0.outlined:hover {
  background: #04ACF4;
  color: white;
}

.c0.disabled {
  background: grey;
}

.c0:hover {
  background: #0484f4;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0:hover.disabled {
  background: grey;
  cursor: default;
}

<button
  class="  c0"
/>
`;

exports[`renders <Button /> with style 1`] = `
.c0 {
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.outlined {
  background: white;
  border: 1px solid #04ACF4;
  color: #04ACF4;
}

.c0.outlined:hover {
  background: #04ACF4;
  color: white;
}

.c0.disabled {
  background: grey;
}

.c0:hover {
  background: #0484f4;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0:hover.disabled {
  background: grey;
  cursor: default;
}

<button
  class="  c0"
  style="float:right"
/>
`;

exports[`renders outlined <Button /> 1`] = `
.c0 {
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.outlined {
  background: white;
  border: 1px solid #04ACF4;
  color: #04ACF4;
}

.c0.outlined:hover {
  background: #04ACF4;
  color: white;
}

.c0.disabled {
  background: grey;
}

.c0:hover {
  background: #0484f4;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0:hover.disabled {
  background: grey;
  cursor: default;
}

<button
  class="outlined  c0"
/>
`;
